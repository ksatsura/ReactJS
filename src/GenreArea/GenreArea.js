//imports
import React from 'react';
import '../style.css';

export const GenreArea = (props) => {

    return <p className='asset-genre'>Films by {props.genre} genre</p>;
};

export default GenreArea;