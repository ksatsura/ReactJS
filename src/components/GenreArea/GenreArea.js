//imports
import React from 'react';
import '../../style.css';

// @flow
type Props = {
    genre: string
};

export const GenreArea = (props) => {

    return <p className='film-genre'>Films by {props.genre} genre</p>;
};

export default GenreArea;