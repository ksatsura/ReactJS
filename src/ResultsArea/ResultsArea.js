//imports
import * as React from 'react';
import FilterButton from '../FilterButton/FilterButton';
import GenreArea from '../GenreArea/GenreArea';
import { Switch, Route } from 'react-router-dom';
import styled from 'styled-components';
import injectSheet from 'react-jss';
import '../style.css';

const ResultsAreaDiv = styled.div`
    width: 1200px;
    height: 60px;
    margin: auto;
    background-color: #e6e6e6;
    display: flex;
    justify-content: space-around;
    align-items: center;
`;

const styles = {
    resultsFilter: {
        display: 'flex',
        width: 250,
        height: 20,
        alignItems: 'baseline',
        justifyContent: 'space-around',
    },
    resultsFilterTitle: {
        fontWeight: 'bold',
        fontSize: 20,
    }
};
  
const ResultsFilter = ({ classes, children }) => (
    <div className={classes.resultsFilter}>
        <p className={classes.resultsFilterTitle}>Sort by</p>
        { children }
    </div>
);

const StyledResultsFilter = injectSheet(styles)(ResultsFilter);

// @flow
type Props = {
    filmsNumber: number, 
    genre: string,
};

const ResultsArea = (props: Props) => {

    const { filmsNumber, genre} = props;
    const content = !filmsNumber && <div className='results-area' id='no-films' ></div>;

    if (content) {
        return content;
    }

    return (
        <ResultsAreaDiv>
            <Switch>
                <Route path='/film/:id' render={() => <GenreArea genre={genre} />} />
                <Route path='/' render={() => <p className='results-title'>{filmsNumber + ' movies found'}</p>} />
            </Switch>
            <Switch>
                <Route path='/film/:id' render={() => <div></div>} /> 
                <Route path='/' render ={() => (<StyledResultsFilter>
                    <FilterButton id='release_date' name='release date' />
                    <FilterButton id='vote_average' name='rating' />
                </StyledResultsFilter>)} />
            </Switch>
        </ResultsAreaDiv>
        
    );
};

export default ResultsArea;