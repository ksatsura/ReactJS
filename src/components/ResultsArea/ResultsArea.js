//imports
import * as React from 'react';
import FilterButton from '../FilterButton/FilterButton';
import GenreArea from '../GenreArea/GenreArea';
import { Switch, Route } from 'react-router-dom';
import styled from 'styled-components';
import injectSheet from 'react-jss';
import '../../style.css';

const ResultsAreaDiv = styled.div`
    width: 1200px;
    height: 60px;
    margin: auto;
    background-color: #e6e6e6;
    display: flex;
    justify-content: space-around;
    align-items: center;
`;

const styles = {
    resultsFilter: {
        display: 'flex',
        width: 250,
        height: 20,
        alignItems: 'baseline',
        justifyContent: 'space-around',
    },
    resultsFilterTitle: {
        fontWeight: 'bold',
        fontSize: 20,
    }
};
  
const ResultsFilter = ({ classes, children }) => (
    <div className={classes.resultsFilter}>
        <p className={classes.resultsFilterTitle}>Sort by</p>
        { children }
    </div>
);

const StyledResultsFilter = injectSheet(styles)(ResultsFilter);

// @flow
type Props = {
    filmsNumber: number, 
    genre: string,
};

const ResultsArea = (props: Props) => {

    const { filmsNumber, genre} = props;
    let content = (filmsNumber === 0) && <div className='results-area' id='no-films' ></div>;

    if (content) {
        return content;
    } else if (genre) {
        content = (
            <ResultsAreaDiv>
                <GenreArea genre={genre} />
                <div></div>
            </ResultsAreaDiv>
        );
    } else {
        content = (<ResultsAreaDiv>
            <p className='results-title'>{filmsNumber + ' movies found'}</p>
            <StyledResultsFilter>
                <FilterButton id='release_date' name='release date' />
                <FilterButton id='vote_average' name='rating' />
            </StyledResultsFilter>
        </ResultsAreaDiv>);
    }

    return content;
};

export default ResultsArea;