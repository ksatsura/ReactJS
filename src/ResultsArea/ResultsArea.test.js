import React from 'react';
import ResultsArea from './ResultsArea';
import GenreArea from '../GenreArea/GenreArea';
import { shallow } from 'enzyme';

describe('ResultsArea.js', () => {

    const props = {
        filmsNumber: 1, 
        isFilmClicked: false, 
        genre: '',
    };

    describe('renders', () => {

        it('if the film was not clicked', () => {
            const component = shallow(<ResultsArea {...props} />);

            expect(component).toMatchSnapshot();
            expect(component.find(GenreArea).length).toEqual(0);
        });

        it('when the film was clicked', () => { 
            const otherProps = {
                ...props,
                isFilmClicked: true,
            };
            const component = shallow(<ResultsArea {...otherProps} />);

            expect(component).toMatchSnapshot();
            expect(component.find(GenreArea).length).toEqual(1);
        });

        it('when no films number', () => {
            const otherProps = {
                ...props,
                filmsNumber: 0,
            };
            const component = shallow(<ResultsArea {...otherProps} />);

            expect(component).toMatchSnapshot();
            expect(component.find('#no-films').length).toEqual(1);
        });
    });
});
