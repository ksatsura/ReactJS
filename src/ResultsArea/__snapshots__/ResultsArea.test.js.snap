// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsArea.js renders if the film was not clicked 1`] = `
<div
  className="results-area"
>
  <p
    className="results-title"
  >
    1 movies found
  </p>
  <div
    className="results-filter"
  >
    <p
      className="results-filter-title"
    >
      Sort by
    </p>
    <Connect(FilterButton)
      id="release_date"
      name="release date"
    />
    <Connect(FilterButton)
      id="vote_average"
      name="rating"
    />
  </div>
</div>
`;

exports[`ResultsArea.js renders when no films number 1`] = `
<div
  className="results-area"
  id="no-films"
/>
`;

exports[`ResultsArea.js renders when the film was clicked 1`] = `
<div
  className="results-area"
>
  <GenreArea
    genre=""
  />
  <div />
</div>
`;
